pkg/kube/util/reference/reconciles.go

```diff
@@ -63,14 +63,14 @@ func GetReconciles(ctx context.Context, client crc.Client, reconcileType Reconci
 			return false, errors.Wrap(err, "error listing references")
 		}
 
-		if versionedSecret && reconcileType != ReconcileForExtendedStatefulSet {
+		if versionedSecret {
 			keys := make([]string, len(objectReferences))
 			i := 0
 			for k := range objectReferences {
 				keys[i] = k
 				i++
 			}
-			ok := vss.ContainsSecretName(keys, name)
+			ok := vss.ContainsOutdatedSecretName(keys, name)
```

Comment:
could we add some comments that we should reconcile if the versioned secret has bumped? Just for readability.

Reply:
done